src/api/middleware/company.js

```diff
@@ -94,7 +94,32 @@ const isNotBlocked = (owner) => async (req, res, next) => {
             ValidationReasons.COMPANY_BLOCKED
         ));
     }
+    return next();
+};
 
+const isNotDisabled = (owner) => async (req, res, next) => {
+    const company = await (new CompanyService()).findById(owner, true);
+    if (company.isDisabled) {
+        return next(new APIError(
+            HTTPStatus.FORBIDDEN,
+            ErrorTypes.FORBIDDEN,
+            ValidationReasons.COMPANY_DISABLED
+        ));
+    }
+    return next();
+};
+
+const canModify = (companyId) => async (req, res, next) => {
+    const company = await (new CompanyService()).findById(companyId, true);
+
+    // only god or the same company can disable a given company
+    if (!req.hasAdminPrivileges && company._id.toString() !== req.targetOwner) {
+        return next(new APIError(
+            HTTPStatus.FORBIDDEN,
+            ErrorTypes.FORBIDDEN,
+            ValidationReasons.COMPANY_NOT_MODIFIABLE
```

Comment:
Please change the error message as well (not only the variable)

```suggestion
            ValidationReasons.INSUFFICIENT_PERMISSIONS_COMAPNY_VISIBILITY
```